### PR TITLE
Set post list data and timestamp based on the user timezone if available

### PR DESCRIPTION
**Pull request by enahum**
#### Summary
Update post timestamp and post list date separators based on user timezone

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-16387

### CODE REVIEW

File: app/components/formatted_time.js [hunk 1]
```diff
@@ -29,11 +29,8 @@ export default class FormattedTime extends React.PureComponent {
         } = this.props;
 
         if (timeZone) {
-            return intl.formatDate(moment.tz(value, timeZone).toDate(), {
```
Comment (jespino) [inline app/components/formatted_time.js]: We were translating it before, why are we no longer translating it?

Comment (enahum) [reply app/components/formatted_time.js]: @jespino the translations where only for PM/AM that is some languages is am/pm but is not massively important and the thing is that formatDate uses the underline `intl` library that does not have support for timeZones

Comment (jespino) [reply app/components/formatted_time.js]: Ok if is on purpose, works for me :)

Comment (jespino) [reply app/components/formatted_time.js]: Ok if is on purpose, works for me :)

**Pull request merged**